File: parking-lot-service-main/src/main/java/com/amanboora/parking/model/ParkingBlock.java
package com.amanboora.parking.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name="parking_block")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ParkingBlock {

    @Id
    private String id;

    @Column(name="parking_id")
    private String parkingId;

    @Column(name="sequence")
    private Long blockSequence;

    @Column(name="name")
    private String blockName;

    @Column(name="parent_block_id")
    private String parentBlockId;

}

File: parking-lot-app-main/src/app/model/parking-block.ts
import {ParkingSpot} from "./parking-spot";

export class ParkingBlock{
    id?:string;
    parkingId?:string;
    parentBlockId?:string;
    sequence: number = 0;
    name?: string;
    blocks?: ParkingBlock[];
    spots?: ParkingSpot[];
    totalSpots:number=0;
    occupiedSpots:number=0;
}
